$NetBSD: patch-configure.ac,v 1.2 2020/01/14 14:44:50 pho Exp $

Hunk #0:
  Don't use non-portable operator ==.


Hunk #1:
  Fix build on FreeBSD/amd64.

--- configure.ac.orig	2018-10-13 19:49:38.000000000 +0000
+++ configure.ac
@@ -914,7 +914,7 @@ FP_CHECK_SIZEOF_AND_ALIGNMENT(uint64_t)
 
 dnl for use in settings.in
 TargetWordSize=$ac_cv_sizeof_void_p
-if test "x$TargetWordSize" == 8; then
+if test "x$TargetWordSize" = 8; then
   AC_SUBST([Cabal64bit],[True])
 else
   AC_SUBST([Cabal64bit],[False])
@@ -1172,6 +1172,11 @@ if test "$ac_cv_sizeof_void_p" -eq 8 ; then
 	    # as of OpenBSD 5.8 (2015), OpenBSD does not support mmap with MAP_NORESERVE.
 	    # The flag MAP_NORESERVE is supported for source compatibility reasons,
 	    # but is completely ignored by OS mmap
+            use_large_address_space=no
+        elif test "$ghc_host_os" = "freebsd" ; then
+            # FreeBSD doesn't support MAP_NORESERVE. The flag
+            # MAP_NORESERVE is supported for source compatibility
+            # reasons, but is completely ignored by OS mmap
             use_large_address_space=no
         else
             AC_CHECK_DECLS([MAP_NORESERVE, MADV_FREE, MADV_DONTNEED],[],[],
